touchforms/backend/touchcare.py

```diff
@@ -31,6 +36,171 @@
+                root = fixture.getRoot()
+            else:
+                root = self._get_fixture(user_id, fixture_id)
+            root.setParent(instance.getBase())
+            return root
+        elif 'ledgerdb' in ref:
+            if settings.USES_SQLITE:
+                ledger_storage = self.sandbox.getLedgerStorage()
+            else:
+                ledger_storage = LedgerDatabase(
+                    self.vars.get('host'), self.vars['domain'],
+                    self.auth, self.vars.get("additional_filters", {}),
+                    self.vars.get("preload_cases", False),
+                )
+            return LedgerInstanceTreeElement(
+                instance.getBase(),
+                ledger_storage
+            )
+
+        elif 'session' in ref:
+            meta_keys = ['device_id', 'app_version', 'username', 'user_id']
+            exclude_keys = ['additional_filters', 'user_data']
+            sess = CommCareSession(None) # will not passing a CCPlatform cause problems later?
+            for k, v in self.vars.iteritems():
+                if k not in meta_keys \
+                        and k not in exclude_keys:
+                    # com.xhaus.jyson.JysonCodec returns data as byte strings
+                    # in unknown encoding (possibly ISO-8859-1)
+                    sess.setDatum(k, unicode(v, errors='replace'))
+
+            clean_user_data = {}
+            for k, v in self.vars.get('user_data', {}).iteritems():
+                clean_user_data[k] = unicode(v if v is not None else '', errors='replace')
+
+            return from_bundle(sess.getSessionInstance(*([self.vars.get(k, '') for k in meta_keys] + \
+                                                         [to_hashtable(clean_user_data)])))
+
+"""
+process_form_file and process_form_xml both perform submissions of the completed form form_data
+against the sandbox of the current user.
+
+"""
+
+
+def process_form_file(auth, submission_file, session_data=None):
+    ccInstances = CCInstances(session_data, auth)
+    sandbox = ccInstances.sandbox
+    FormRecordProcessor.processFile(sandbox, File(submission_file))
+
+
+def process_form_xml(auth, submission_xml, session_data=None):
+    ccInstances = CCInstances(session_data, auth)
+    sandbox = ccInstances.sandbox
+    FormRecordProcessor.processXML(sandbox, submission_xml)
+
+
+def perform_restore(auth, session_data=None, restore_xml=None):
+    CCInstances(session_data, auth, restore_xml, True)
+
+
+def filter_cases(filter_expr, auth, session_data=None, restore_xml=None, needs_sync=True):
+
+    modified_xpath = "join(',', instance('casedb')/casedb/case%(filters)s/case_name)" % \
+        {"filters": filter_expr}
+
+    ccInstances = CCInstances(session_data, auth, restore_xml, needs_sync)
+    caseInstance = ExternalDataInstance("jr://instance/casedb", "casedb")
+    caseInstance.initialize(ccInstances, "casedb")
```

Comment:
i know i told you to get rid of the try catch before but now that we have old code back in that makes web requests we may want to keep it until we fully convert